### PR TITLE
feat: be asynchronous

### PR DESCRIPTION
**Pull request by PaperStrike**
Will be a breaking change but seem to be the only way to support ESLint 8. 😬

May need some new tests for cli coverage.

cc @ljharb

### CODE REVIEW

**Comment by PaperStrike** [comment]
Parent: https://github.com/sarbbottam/eslint-find-rules/pull/337

File: src/bin/find.js [hunk 1]
```diff
@@ -33,13 +33,15 @@ const getRuleURI = require('eslint-rule-documentation');
 const getRuleFinder = require('../lib/rule-finder');
 const cli = require('../lib/cli-util');
 
+module.exports = (async function() {
```
Comment (ljharb) [inline src/bin/find.js:36]:
```suggestion
(async function () {
```
bin files shouldn't export anything

File: .babelrc [hunk 2]
```diff
@@ -1,3 +1,10 @@
...
-  "presets": ["es2015"]
+  "presets": [
+    ["env", {
+      "targets": {
+        "node": "4"
+      },
+      "exclude": ["transform-regenerator"]
+    }]
+  ]
 }
```
Comment (ljharb) [inline .babelrc:10]: this is a good change; can you make a separate PR to the main repo switching to the env preset?

File: src/bin/find.js (hunk 1 above)
Comment (PaperStrike) [reply src/bin/find.js:36]: It's for the test cases. The test cases just import them as modules instead of execute in child processes. E.g.,

https://github.com/MichaelDeBoey/eslint-find-rules/blob/d42b2ec8dac31b692aeab8a49c6293c97fa2dfeb/test/bin/find.js#L66-L71

Would need a lot of changes if we don't export this way. 😬

File: .babelrc (hunk 2 above)
Comment (PaperStrike) [reply .babelrc:10]: ok. Filled https://github.com/sarbbottam/eslint-find-rules/pull/340

Comment (PaperStrike) [reply .babelrc:10]: will start rebase after https://github.com/sarbbottam/eslint-find-rules/pull/337. 👀

File: src/bin/find.js (hunk 1 above)
Comment (PaperStrike) [reply src/bin/find.js:36]: Ah... I don't know any package or manual way that can mock in child processes, and keep the coverage data at the same time. They seem hard. Any idea?

Comment (ljharb) [reply src/bin/find.js:36]: I agree, that's difficult. How is this file tested now?

Comment (PaperStrike) [reply src/bin/find.js:36]: it's being tested like normal modules. its test file changes `process.argv` to simulate cli env, and mocks`process.exit` `console.log` and `rule-finder.js` to test the output.

Comment (ljharb) [reply src/bin/find.js:36]: oof, that's also gross and hacky.

I'd rather move the logic out of the bin file (in a different PR to the source repo) and make that test cleaner, and then the changes here will be smaller.

Comment (PaperStrike) [reply src/bin/find.js:36]: So we can't run actual child processes. Trying to write explicit `process.exit`s and add their listeners to find out when the cli scripts are finished.

Comment (PaperStrike) [reply src/bin/find.js:36]: I fell into the trap of child processes, thinking another process is the only way to tell when it is finished. Now I'm out. Commit later. 😉

Comment (PaperStrike) [reply src/bin/find.js]: > I'd rather move the logic out of the bin file (in a different PR to the source repo) and make that test cleaner, and then the changes here will be smaller.

Yes, I suggest we find or write a tool that wraps these. 👀 That will be much better.

Comment (ljharb) [reply src/bin/find.js]: What I mean is, take the "bin" file entirely out of the testing equation, and make all the non-bin stuff be tested normally.

Comment (PaperStrike) [reply src/bin/find.js]: Oh. Sorry I didn't read it carefully.

Could you finish this for me?

Comment (PaperStrike) [reply src/bin/find.js]: I'll give a try later.

Comment (PaperStrike) [reply src/bin/find.js]: Done! I must be drunk that everything easy seems so difficult at the moment.

Comment (PaperStrike) [reply src/bin/find.js]: Marking as resolved now. Future conversation for this goes to https://github.com/MichaelDeBoey/eslint-find-rules/pull/14.

**Comment by PaperStrike** [comment]
@ljharb Thank you! This PR changes now looks so clean!

There should be no confliction when merging this and https://github.com/MichaelDeBoey/eslint-find-rules/pull/14, since they change different files.
